stan/math/prim/fun/eigenvectors_sym.hpp

```diff
@@ -12,10 +12,13 @@ template <typename EigMat, require_eigen_t<EigMat>* = nullptr,
           require_not_st_var<EigMat>* = nullptr>
 Eigen::Matrix<value_type_t<EigMat>, Eigen::Dynamic, Eigen::Dynamic>
 eigenvectors_sym(const EigMat& m) {
+  if (unlikely(m.size() == 0)) {
+    return Eigen::Matrix<value_type_t<EigMat>, -1, -1>(0, 0);
+  }
+  check_symmetric("eigenvalues_sym", "m", m);
```

Comment:
`check_symmetric` does elementwise access so need to use `to_ref` here
```suggestion
  auto&& m_ref = to_ref(m)
  check_symmetric("eigenvalues_sym", "m", m_ref);
```